### DISCUSSION TITLE
Ray datasets

### DISCUSSION THREAD

**Discussion by SumanthDatta-Kony** [ludwig-ai/ludwig#1518] [Q&A]
I have taken the code base from git , trying to use Ray datasets from ludwig.data.dataset.ray package. So basically trying to pass ray dataset pipeline for training.

Getting below exception:
Cannot figure out the format of dataset {}".format(dataset)
ValueError: Cannot figure out the format of dataset <ludwig.data.dataset.ray.RayDatasetShard object at 0x7ff5e97abed0>

Is this fully implemented  or work in progress? , if it is implemented can you provide me with a sample example on how to use ray datasets. Any help is appreciated.

Thanks

Sumanth

**Comment by tgaddair** [comment]
Hey @SumanthDatta-Kony, instead of passing the Ray Dataset directly to Ludwig, can you convert it into a Dask DataFrame first using `to_dask`? We don't yet support passing a Ray Dataset as-is, but this would be a relatively simple thing to add (by doing the to_dask conversion on our end).
